single_res[] single_res_arrary
